### PR TITLE
[TEMPLATE] GridSample in template plugin supports more precisions

### PR DESCRIPTION
**Pull request by praasz**
### Details:
 - Add more precisions support for `GridSample` in Template plugin.

### Tickets:
 - [CVS-133057](https://jira.devtools.intel.com/browse/CVS-133057)

### CODE REVIEW

File: src/core/reference/include/openvino/reference/grid_sample.hpp [hunk 1]
```diff
@@ -139,15 +139,14 @@ DATA_ET bilinear(const DATA_ET* data,
...
-    const auto dx = x_d - x_topleft;
-    const auto v00 = get_padded(data, data_shape, n, c, static_cast<long>(y_topleft), static_cast<long>(x_topleft));
-    const auto v01 = get_padded(data, data_shape, n, c, static_cast<long>(y_topleft), static_cast<long>(x_topleft + 1));
-    const auto v10 = get_padded(data, data_shape, n, c, static_cast<long>(y_topleft + 1), static_cast<long>(x_topleft));
-    const auto v11 =
-        get_padded(data, data_shape, n, c, static_cast<long>(y_topleft + 1), static_cast<long>(x_topleft + 1));
+    const auto y_topleft = static_cast<long>(std::floor(y_d));
+    const auto x_topleft = static_cast<long>(std::floor(x_d));
+    const auto dy = static_cast<DATA_ET>(y_d - y_topleft);
+    const auto dx = static_cast<DATA_ET>(x_d - x_topleft);
```
Comment (t-jankowski) [inline src/core/reference/include/openvino/reference/grid_sample.hpp:145]: (`GRID_ET` - `long`) casted to `DATA_ET` .. is this `long` really needed here? What's the promotion of implicit cast when `GRID_ET` is non `std::` floating point type?

Comment (praasz) [reply src/core/reference/include/openvino/reference/grid_sample.hpp:145]: It is not required, restored previous version.

**Review by t-jankowski — APPROVED** [review]

File: src/core/reference/include/openvino/reference/grid_sample.hpp (hunk 1 above)
Comment (vurusovs) [reply src/core/reference/include/openvino/reference/grid_sample.hpp:145]: Should the change be reverted?

File: src/plugins/template/backend/ops/grid_sample.cpp [hunk 2]
```diff
@@ -34,23 +64,17 @@ template <>
 bool evaluate_node<ov::op::v9::GridSample>(std::shared_ptr<ov::Node> node,
                                            ov::TensorVector& outputs,
                                            const ov::TensorVector& inputs) {
-    auto element_type = node->get_output_element_type(0);
-    if (ov::is_type<ov::op::v1::Select>(node) || ov::is_type<ov::op::util::BinaryElementwiseComparison>(node))
-        element_type = node->get_input_element_type(1);
```
Comment (vurusovs) [inline src/plugins/template/backend/ops/grid_sample.cpp:39]: Why the change not required anymore?

Comment (praasz) [reply src/plugins/template/backend/ops/grid_sample.cpp:39]: Is not required as this evaluate is registered for `v9::GridSample` and this check will be false.

**Review by vurusovs — APPROVED** [review]
